Adds back,edit,promotions buttons in profile business page and edit functionality

LogicX1 opened the pull request:
Added the button names in both languages
Changed choose radius to distance, and added it in arabic as well
some basic styling
rel #59 #52

LogicX1 committed:
Adds back,edit,promotions buttons in profile business page
Added the button names in both languages
Changed choose radius to distance, and added it in arabic as well
some basic styling
rel #59 #52

LogicX1 committed:
Adds functionality to edit business
Adds a selected business state
Adds a editBusiness state which indicates wether we are editing (will have the business id) or null otherwise
rel #52 #53

LogicX1 committed:
Adds more functionality
Now we store the whole business instead of only the id, because this data is needed when we are going to edit the business
Modified small things here and there

LogicX1 committed:
Improves draggable pin to choose location in add business
During testing it was very annoying to drag the pin initially , now its better
rel #63

Karyum committed:
resolve conflcit

Karyum committed:
Merge branch 'master' into profile-business-page

LogicX1 committed:
More work on the edit business
It is not finished yet ,Ill piick this up again asap
I need to have a state on addbusiness to know wethere im editing or adding new one
Not all feilds are filled ,text ones are but not the select type ones

LogicX1 committed:
Merge branch 'edit-business' of https://github.com/WebAhead/wajjbat-frontend into profile-business-page

LogicX1 commented:
This needs a bit more work, will finish it soon

rel#52 #53

LogicX1 committed:
Merge branch 'master' into profile-business-page

Karyum commented:
@LogicX1 the Netlify check is failing, meaning the build script for the frontend is failing

LogicX1 committed:
Fixing an error that stopped netlify check
The problem occured when the merge, the result lacked a comma between what I added and the others added, I think thats why the dangling comma is a good rule to have

LogicX1 committed:
Added dangling commas

LogicX1 committed:
Fixing git error

LogicX1 committed:
Changes styles for the new buttons,check boxes and list items are now autofilled,fixed errors and react warnings
rel #43 #52 #53

LogicX1 committed:
Fixes a small mistake from the previous commit

LogicX1 committed:
Fixes gettings parking,Free Wifi and Smoking Area
The get business by id from the backend did not return those feilds, its added there and here changed the relavent variable names

LogicX1 committed:
Adds picture links to state when editing

LogicX1 committed:
Adds basic form validation and fixes default value on select
Now if you dont upload an image,or do not select type or cuisine, the submit will not go throgh, shows relevant messages to the user as well

LogicX1 committed:
Adds the option to send to the update api route when editing,error handling,back functionality after query succesful
rel #74 #52 #53

LogicX1 commented:
Things took longer than expected, there were some warning fixes, bug fixes and some error handling that took extra work, did not get to finish the part with images, will finish it tomorrow.
